### PR TITLE
Make test_ssl pass in an IPv6-only environment

### PR DESCRIPTION
**Pull request by davidben**

### CODE REVIEW

File: tests/test_ssl.py [hunk 1]
```diff
@@ -101,6 +101,21 @@
 skip_if_py3 = pytest.mark.skipif(PY3, reason="Python 2 only")
 
 
+def socket_any_family():
+    try:
+        return socket(AF_INET)
+    except error:
```
Comment (alex) [inline tests/test_ssl.py]: I'm sure I'll regret asking, is there any way for us to narrow the exception down so we don't handle other failures?

File: tests/test_ssl.py [hunk 2]
```diff
@@ -101,6 +101,21 @@
...
+        return socket(AF_INET6)
+
+
+def loopback_address(socket):
+    if socket.family == AF_INET:
+        return "127.0.0.1"
+    if socket.family == AF_INET6:
+        return "::1"
+    raise ValueError("Unknown socket family")
```
Comment (alex) [inline tests/test_ssl.py]: To make coverage happy you can write this:

```
if socket.family == AF_INET:
    return "127.0.0.1"
else:
    assert socket.family == AF_INET6
    return "::1"
```

my kingdom for proper enums 🤷‍♂️

File: tests/test_ssl.py (hunk 1 above)
Comment (alex) [reply tests/test_ssl.py]: Yes, this should have an `errno` attribute, let's check that it's the right one.

Comment (davidben) [reply tests/test_ssl.py]: I take it Python on Windows will try to map the errno and such?

Comment (alex) [reply tests/test_ssl.py]: I think so. I've definitely written a lot of code that does `.errno == ENOENT` and no one has complained 😬

Comment (davidben) [reply tests/test_ssl.py]: Done.

File: tests/test_ssl.py (hunk 2 above)
Comment (davidben) [reply tests/test_ssl.py]: Done.

File: tests/test_ssl.py [hunk 3]
```diff
@@ -101,6 +101,23 @@
 skip_if_py3 = pytest.mark.skipif(PY3, reason="Python 2 only")
 
 
+def socket_any_family():
+    try:
+        return socket(AF_INET)
+    except error as e:
+        if e.errno == EAFNOSUPPORT:
+            return socket(AF_INET6)
+        raise e
```
Comment (alex) [inline tests/test_ssl.py]:
```suggestion
        raise
```

this will preserve the stack trace better.

File: tests/test_ssl.py [hunk 4]
```diff
@@ -10,9 +10,9 @@
 import uuid
 
 from gc import collect, get_referrers
-from errno import ECONNREFUSED, EINPROGRESS, EWOULDBLOCK, EPIPE, ESHUTDOWN
+from errno import EAFNOSUPPORT, ECONNREFUSED, EINPROGRESS, EWOULDBLOCK, EPIPE, ESHUTDOWN
```
Comment (alex) [inline tests/test_ssl.py]: Eeep, line too long, can you wrap this with `()`

Comment (davidben) [reply tests/test_ssl.py]: Done.

**Pull request merged**